docs/modules/operation/pages/provisioning/detectors/BsfDetector.adoc

```diff
@@ -0,0 +1,175 @@
+
+== Requirements for the script (run-types)
+
+Depending on the `run-type` the script has to provide its results in different ways.
+For minimal scripts with very simple logic `run-type` `eval` is the simple option.
+Scripts running in `eval` mode have to return a _String_ matching one of the `status codes`.
+
+If your script is more than a one-liner, `run-type` `exec` is essentially required.
+Scripts running in `exec` mode need not return anything, but they have to add a `status` entry with a `status code` to the _results_ object.
+Additionally, the _results_ object can also carry a "reason":"message" entry that is used in non `OK` states.
+
+== Commonly used language settings
+
+The _BSF_ supports many languages, the following table provides the required setup for commonly used languages.
+
+._BSF_ language setups
+[options="header, autowidth"]
+|===
+| Language                            | lang-class  | bsf-engine                                    | required library
+| http://www.beanshell.org[BeanShell] | _beanshell_ | `bsh.util.BeanShellBSFEngine`                 | supported by default
+| http://groovy.codehaus.org[Groovy]  | _groovy_    | `org.codehaus.groovy.bsf.GroovyEngine`        | +groovy-all-[version].jar+
+| http://www.jython.org[Jython]       | _jython_    | `org.apache.bsf.engines.jython.JythonEngine`  | +jython-[version].jar+
+|===
+
+
+== Example Configurations
+
+=== Example Bean Shell
+
+._BeanShell_ example
+[source, xml]
+----
+<detector name="MinimalBeanShell" class="org.opennms.netmgt.provision.detector.bsf.BSFDetector">
+    <parameter key="bsfEngine" value="bsh.util.BeanShellBSFEngine"/>
+    <parameter key="langClass" value="beanshell"/>
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalBeanShell.bsh"/>
+    <parameter key="runType" value="eval"/>
+</detector>
+----
+
+.BeanShell example `MinimalBeanShell.bsh` script file
+[source, java]
+----
+File testFile = new File("/tmp/TestFile");
+if (testFile.exists()) {
+  return "OK";
+} else {
+  results.put("reason", "file does not exist");
+  return "NOK";
+}
+----
+
+=== Example Groovy
+
+._Groovy_ example `MinimalGroovy.groovy` script file for `run-type` `eval`
+[source, xml]
+----
+<detector name="MinimalGroovy" class="org.opennms.netmgt.provision.detector.bsf.BSFDetector">
+    <parameter key="bsfEngine" value="org.codehaus.groovy.bsf.GroovyEngine"/>
+    <parameter key="langClass" value="groovy"/>
+    <parameter key="fileName" value="/opt/opennms/etc/scripts/MinimalGroovy.groovy"/>
+    <parameter key="runType" value="eval"/>
+</detector>
+----
+
+.BeanShell example `MinimalGroovy.groovy` script file for `run-type` `eval`
+[source, java]
+----
+File testFile = new File("/tmp/TestFile");
+if (testFile.exists()) {
+  return "OK";
+} else {
+  results.put("reason", "file does not exist");
+  return "NOK";
+}
+----
+
+._Groovy_ example `MinimalGroovy.groovy` script file for `run-type` `exec`
```

Comment:
```suggestion
._Groovy_ example for `run-type` `exec`
```

Reply:
@mfuhrmann . Strange. Just need to remove the italics around "Groovy".